core/templates/domain/topic_viewer/read-only-topic-object.factory.spec.ts

```diff
@@ -0,0 +1,132 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for read-only-topic-object.factory.ts
+ */
+import { TestBed } from '@angular/core/testing';
+
+import { ReadOnlyTopic, ReadOnlyTopicObjectFactory } from
+  'domain/topic_viewer/read-only-topic-object.factory';
+import { SkillSummaryObjectFactory } from
+  'domain/skill/SkillSummaryObjectFactory';
+import { SubtopicObjectFactory } from
+  'domain/topic/SubtopicObjectFactory';
+
+describe('Topic Data Object Factory', () => {
+  let readOnlyTopicObjectFactory: ReadOnlyTopicObjectFactory = null;
+  let _sampleReadOnlyTopic: ReadOnlyTopic = null;
+
+  beforeEach(() => {
+    readOnlyTopicObjectFactory = new ReadOnlyTopicObjectFactory(
+      new SubtopicObjectFactory(new SkillSummaryObjectFactory()),
+      new SkillSummaryObjectFactory());
+
+    let sampleTopicDataDict = {
+      topic_name: 'topic_name',
+      topic_id: 'topic_id',
+      canonical_story_dicts: [{
+        id: '0',
+        title: 'Story Title',
+        description: 'Story Description',
+        node_count: 1,
+        published: true
+      }],
+      additional_story_dicts: [{
+        id: '1',
+        title: 'Story Title',
+        description: 'Story Description',
+        node_count: 1,
+        published: true
+      }],
+      uncategorized_skill_ids: ['skill_id_1'],
+      subtopics: [{
+        skill_ids: ['skill_id_2'],
+        id: 1,
+        title: 'subtopic_name'
+      }],
+      degrees_of_mastery: {
+        skill_id_1: 0.5,
+        skill_id_2: 0.3
+      },
+      skill_descriptions: {
+        skill_id_1: 'Skill Description 1',
+        skill_id_2: 'Skill Description 2'
+      },
+      train_tab_should_be_displayed: false
+    };
+
+    _sampleReadOnlyTopic = readOnlyTopicObjectFactory.createFromBackendDict(
+      sampleTopicDataDict);
+  });
+
+  it('should check the values of topic name and id', () => {
+    expect(_sampleReadOnlyTopic.getTopicName()).toEqual('topic_name');
+    expect(_sampleReadOnlyTopic.getTopicId()).toEqual('topic_id');
+    expect(_sampleReadOnlyTopic.getTrainTabShouldBeDisplayed()).toEqual(false);
+  });
+
+  it('should check value of uncategorized skill object', () => {
+    expect(_sampleReadOnlyTopic.getUncategorizedSkillsSummaries()[0].getId())
+      .toEqual('skill_id_1');
+    expect(_sampleReadOnlyTopic.getUncategorizedSkillsSummaries()[0].
+      getDescription()).toEqual('Skill Description 1');
+  });
+
+  it('should check values of Subtopic object', () => {
+    expect(_sampleReadOnlyTopic.getSubtopics()[0].getId()).toEqual(1);
+    expect(_sampleReadOnlyTopic.getSubtopics()[0].getTitle()).toEqual(
+      'subtopic_name');
+    expect(_sampleReadOnlyTopic.getSubtopics()[0]._skillSummaries[0].getId()).
+      toEqual('skill_id_2');
+    expect(_sampleReadOnlyTopic.getSubtopics()[0]._skillSummaries[0].
+      getDescription()).toEqual('Skill Description 2');
+  });
+
+  it('should check type and values of Skill Description', () => {
```

Comment:
instead of `check` I would suggest `return correct ...` (here and elsewhere)

Reply:
ok